### PR TITLE
fix(shellcheck): detect RUN heredoc opener after continuation-wrapped flags

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

- Fix heredoc-body extraction when `RUN --mount=...` flags span multiple continuation lines before `<<TAG`. Previously the opener detection used `firstNonEmptyLine`, which picked up lines containing only `\` continuations and bailed out — so the whole multi-line span (flags, continuations, `<<EOF`, body, and `EOF`) was fed verbatim to shellcheck, producing false positives.
- Refresh `_tools/shellcheck-wasm/Dockerfile` to exercise the patterns this fix handles: a `COPY --chmod=0644 <<EOF ${CURL_HOME}/.curlrc` heredoc, plus a `RUN --mount=type=cache,...` / `--mount=type=bind,...` / `<<EOF` block split across three continuation lines.
- Switch `make shellcheck-wasm` to `docker buildx build` so the mount flags resolve on all supported BuildKit versions.

## Root cause

`heredocBodyScriptMode` called `firstNonEmptyLine(blanked)` to locate the opener after flag-blanking. Flag-blanking preserves trailing `\` characters on each continuation line, so when flags wrap across lines the "first non-empty" line was just `\` — heredoc detection failed. The caller then fed the full span to shellcheck and flagged:

- **shellcheck/SC2188** at `<<EOF` (seen as a redirection with no command)
- **shellcheck/SC2154** on `$r` and `$rule_id` (loop variables defined inside the body appeared inside opaque heredoc context to shellcheck)
- Wrong columns because the script handed to shellcheck started several columns left of the real body lines.

## Fix

Replaced `firstNonEmptyLine` with `firstHeaderLine`, which skips blank lines and lines that contain only the escape token, and returns the opener's 0-based line offset within the span. The caller now computes `bodyStart := start + openerOffset + 1`, so the body origin line is correct even when the `<<TAG` opener lives several continuation lines past the instruction start.

## Test plan

- [x] `go test ./internal/highlight/extract/...` — added two new tests covering multi-line flags before a plain heredoc and before an explicit-shell heredoc
- [x] `go test ./internal/rules/shellcheck/... ./internal/integration/...` — no regressions
- [x] `./tally lint _tools/shellcheck-wasm/Dockerfile` — SC2188 / SC2154 false positives are gone
- [ ] CI: verify `make shellcheck-wasm` still builds with `docker buildx build`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed heredoc parsing when flags span multiple continued lines in Dockerfile RUN commands.

* **Build & Infrastructure**
  * Switched container image builds to buildx for more reliable builds.
  * Added curl retry/timeouts and npm cache to improve build robustness.

* **Tests**
  * Added tests covering heredoc extraction with continuation-wrapped flags and related edge cases.

### CODE REVIEW

File: internal/highlight/extract/script_test.go [hunk 1]
```diff
@@ -74,6 +74,54 @@ EOF
...
+	if mapping.ShellNameOverride != "bash" {
+		t.Fatalf("expected shell override bash, got %q", mapping.ShellNameOverride)
+	}
+	if mapping.Script != "echo hi" {
+		t.Fatalf("expected body-only script, got %q", mapping.Script)
+	}
+	if mapping.OriginStartLine != 4 {
+		t.Fatalf("expected body origin line 4, got %d", mapping.OriginStartLine)
+	}
+}
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/highlight/extract/script_test.go:123]: ✅ Addressed in 4aa4a2f. Thanks @coderabbitai!

File: internal/highlight/extract/script.go [hunk 2]
```diff
@@ -140,43 +140,60 @@ func extractRunLikeScript(
...
+			continue
+		}
+		if trimmed == escape {
+			continue
 		}
+		return i, line
 	}
-	return ""
+	return 0, ""
 }
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/highlight/extract/script.go:199]: ✅ Addressed in 7b199bc. Thanks @gemini-code-assist!
